nncf/common/graph/transformations/commands.py

```diff
@@ -11,14 +11,17 @@
  limitations under the License.
 """
 
+from typing import Any
+
 from nncf.common.utils.ordered_enum import OrderedEnum
 
 
 class TransformationPriority(OrderedEnum):
     DEFAULT_PRIORITY = 0
+    FP32_TENSOR_STATISTICS_OBSERVATION = 1
     SPARSIFICATION_PRIORITY = 2
     QUANTIZATION_PRIORITY = 11
-    PRUNING_PRIORITY = 1
+    PRUNING_PRIORITY = 3
```

Comment:
How does it work before with the first priority? We didn't check it in combination with other algorithms, did we?

Reply:
I have two concerns which I address by changing PRUNING_PRIORITY from 1 to 3:
- FP32_TENSOR_STATISTICS_OBSERVATION and PRUNING_PRIORITY have the same priority. This can lead to non-deterministic behavior.
- When Pruning and Sparsity algorithms are stacked. My opinion is that the sparsity mask should be calculated for the pruned weights that have already been pruned by the filter pruning algorithm. Otherwise, the sparsity rate for all models in the sparsity algorithm will be incorrect.

This change influences only the interaction of the sparsity and pruning algorithms. I didn't check it in combination. But If you agree with my statement, I propose to prove this change.

Reply:
Sounds reasonable. 
It makes sense to fix [test_priority](https://github.com/openvinotoolkit/nncf/blob/develop/tests/test_nncf_network.py#L257) with new behaviour then.

Reply:
I have added tests. Please take a look.